homeassistant/components/smarttub/sensor.py

```diff
@@ -14,6 +19,25 @@
 # the hour of the day at which to start the cycle (0-23)
 ATTR_START_HOUR = "start_hour"
 
+SET_PRIMARY_FILTRATION_SCHEMA = vol.All(
+    cv.has_at_least_one_key(ATTR_DURATION, ATTR_START_HOUR),
+    cv.make_entity_service_schema(
+        {
+            vol.Optional(ATTR_DURATION): vol.All(int, vol.Range(min=1, max=24)),
+            vol.Optional(ATTR_START_HOUR): vol.All(int, vol.Range(min=0, max=23)),
```

Comment:
According to the service.yaml file, both are required? According to this, at least one key?